Dockerfile

```diff
@@ -0,0 +1,35 @@
+# Copyright (C) Daniel Stenberg, <daniel@haxx.se>, et al.
+#
+# SPDX-License-Identifier: curl
+
+# Self-contained build environment to match the release environment.
+#
+# Build with
+#
+#   docker build -t curl/curl .
+#
+# Then run commands from within the build environment, for example
+#
+#   docker run --rm -v $(pwd):/usr/src -w /usr/src curl/curl autoreconf -fi
+#   docker run --rm -v $(pwd):/usr/src -w /usr/src curl/curl ./configure --without-ssl --without-libpsl
+#   docker run --rm -v $(pwd):/usr/src -w /usr/src curl/curl make
+#   docker run --rm -v $(pwd):/usr/src -w /usr/src curl/curl ./maketgz 8.7.1
+
+FROM debian:bookworm-slim@sha256:993f5593466f84c9200e3e877ab5902dfc0e4a792f291c25c365dbe89833411f
+# autoconf 2.71
+# automake 1.16.5
+# libtoolize 2.4.7
+# make 4.3
+# perl 5.36.0
+# git 2.39.2
+
+RUN apt-get update -qq && apt-get install -qq -y --no-install-recommends \
+    build-essential make autoconf automake libtool git perl zip zlib1g-dev gawk && \
+    rm -rf /var/lib/apt/lists/*
+
+# >>> from datetime import datetime
+# >>> int(datetime(2024, 3, 27, 9).timestamp())
+# 1711526400
+ENV LC_ALL=C TZ=UTC SOURCE_DATE_EPOCH=1711526400
```

Comment:
```suggestion
# >>> from datetime import datetime, timezone
# >>> int(datetime(2024, 3, 27, 23, 54, 59, tzinfo=timezone.utc).timestamp())
# 1711583699
ENV SOURCE_DATE_EPOCH=1711583699
```
Settle TZ and use a timestamp which makes it easier to spot when messing up the timezone.

Also, `maketgz` should take care of `LC_ALL` and `TZ` now, do we still need them here as well?

Reply:
I don't think we should set a fixed timestamp in the Dockerfile. I think we can have the comment there to show *how* to set the timestamp if wanted, but if we leave out the stamp there we can use this file unmodified for doing daily snapshots and the real releases

Reply:
Yes, it's even better.

Or derive it automatically from the git commit date for daily runs. That can also be done outside the `Dockerfile`.

Reply:
Agree with you folks, when building the user needs to supply `SOURCE_DATE_EPOCH`.

I'll get to this when I rebase and also folks in the `LC_ALL` and `TZ` changes that are now in master.

Reply:
It is pretty common to set SOURCE_DATE_EPOCH=1 to simply disable timestamps for all intents and purposes which is what we do in stagex globally so we can reproduce packages at any point in the future from any ref easily.

I suggest `ARG SOURCE_DATE_EPOCH=1` so you can override it at build time with `--build-arg SOURCE_DATE_EPOCH=1234` else it defaults to 1.